src/test/java/org/opensearch/StreamInputOutputTests.java

```diff
@@ -0,0 +1,293 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch;
+
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.emptySet;
+import static org.hamcrest.Matchers.equalTo;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.opensearch.action.FailedNodeException;
+import org.opensearch.ad.AbstractADTest;
+import org.opensearch.ad.model.Entity;
+import org.opensearch.ad.model.EntityProfileName;
+import org.opensearch.ad.model.ModelProfile;
+import org.opensearch.ad.model.ModelProfileOnNode;
+import org.opensearch.ad.transport.EntityProfileAction;
+import org.opensearch.ad.transport.EntityProfileRequest;
+import org.opensearch.ad.transport.EntityProfileResponse;
+import org.opensearch.ad.transport.EntityResultRequest;
+import org.opensearch.ad.transport.ProfileNodeResponse;
+import org.opensearch.ad.transport.ProfileResponse;
+import org.opensearch.ad.transport.RCFResultResponse;
+import org.opensearch.cluster.ClusterName;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.common.io.stream.BytesStreamOutput;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.transport.TransportAddress;
+
+/**
+ * Put in core package so that we can using Version's package private constructor
+ *
+ */
+public class StreamInputOutputTests extends AbstractADTest {
```

Comment:
is the main idea here that it's the same as [bwc](src/main/java/org/opensearch/ad/util/Bwc.java), but removes the tests and conditional statements about 1.x requests/responses? Anything in particular that's been added in this file? It's a lot of lines to try to compare

Reply:
yes, it is almost the same except I added a few more lines of code to boost the test coverage. You can deem it the same.